src/runtime/os_linux.go

```diff
@@ -139,3 +146,16 @@ func hardwareRand() (n uint64, ok bool) {
 //
 //export getrandom
 func libc_getrandom(buf unsafe.Pointer, buflen uintptr, flags uint32) uint32
+
+// int fcntl(int fd, int cmd, int arg);
+//
+//export fcntl
+func libc_fcntl(fd int, cmd int, arg int) (ret int)
+
+func fcntl(fd int32, cmd int32, arg int32) (ret int32, errno int32) {
+	ret = int32(libc_fcntl(int(fd), int(cmd), int(arg)))
+	if ret < 0 {
+		return 0, int32(*libc_errno_location())
+	}
+	return ret, 0
```

Comment:
This is not quite correct. If you look at the original location, the function is used like this:

```go
func Fcntl(fd int, cmd int, arg int) (int, error) {
        val, errno := fcntl(int32(fd), int32(cmd), int32(arg))
        if val == -1 {
                return int(val), syscall.Errno(errno)
        }
        return int(val), nil
}
```

So `val` needs to be -1 when there's an error.
This probably means you can simplify the code like this:

```suggestion
	ret = int32(libc_fcntl(int(fd), int(cmd), int(arg)))
	errno = *libc_errno_location()
	return
```